File: content/3-leetcode/1-two-sum/go/two_sum.go
package main

func twoSum(nums []int, target int) []int {
    numMap := make(map[int]int)
    for i, num := range nums {
        complement := target - num
        if idx, ok := numMap[complement]; ok {
            return []int{idx, i}
        }
        numMap[num] = i
    }
    return nil
}

File: content/3-leetcode/1-two-sum/cs/TwoSum.cs
using System;
using System.Collections.Generic;

public class TwoSumSolution
{
    public static int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> numMap = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++)
        {
            int complement = target - nums[i];
            if (numMap.ContainsKey(complement))
            {
                return new int[] { numMap[complement], i };
            }
            numMap[nums[i]] = i;
        }
        return new int[0];
    }
}